examples/source_examples/veeva_vault_objects/connector.py

```diff
@@ -0,0 +1,171 @@
+    deactivate_session_url = f"https://{configuration.get('vaultDNS')}/api/v24.2/session"
+    headers = {
+    "Authorization": session_id
+    }
+    response_deact = requests.delete(deactivate_session_url, headers=headers)
+    log.info(f"Deactivation of session for user: {configuration.get('username')} responded with code: {response_deact.status_code}: {response_deact.reason}")
+    
+# Function to get list of available Veeva Vault objects
+# Endpoint documentation: https://developer.veevavault.com/api/24.2/#retrieve-details-from-all-object-types
+# The getVaultObjects function takes 2 parameters:
+# - configuration: a dictionary that holds the configuration settings for the connector.
+# - session_id: a Veeva Vault session ID generated from the startVeevaSession function
+def getVaultObjects(configuration: dict, session_id):
+    base_url = f"https://{configuration.get('vaultDNS')}/api/v24.2/"
+    headers = {'Authorization': session_id,
+            'Accept': 'application/json'}
+    get_objects_url = base_url + 'configuration/Objecttype'
+    obj_response = requests.get(get_objects_url, headers = headers).json()
+    objects = obj_response.get('data')
+    return objects
+
+# Define the schema function which lets you configure the schema your connector delivers.
+# See the technical reference documentation for more details on the schema function:
+# https://fivetran.com/docs/connectors/connector-sdk/technical-reference#schema
+# The schema function takes 1 parameter:
+# - configuration: a dictionary that holds the configuration settings for the connector.
+def schema(configuration: dict):
+    # Dynamically define schema based on getVaultObjects response
+    # Note: Defining all tables to have a PK of "id" right now - need to verify that this will hold true for all objects
+    session_id = startVeevaSession(configuration)
+    objects = getVaultObjects(configuration, session_id)
+    schema_def = []
+    for obj in objects:
+        schema_def.append({"table": obj.get('label_plural'), "primary_key": ["id"]})
+    endVeevaSession(configuration, session_id)
+    return schema_def
+
+# Define the update function, which is a required function, and is called by Fivetran during each sync.
+# See the technical reference documentation for more details on the update function
+# https://fivetran.com/docs/connectors/connector-sdk/technical-reference#update
+# The function takes 2 parameters:
+# - configuration: dictionary contains any secrets or payloads you configure when deploying the connector
+# - state: a dictionary contains whatever state you have chosen to checkpoint during the prior sync
+# The state dictionary is empty for the first sync or for any full re-sync
+# This function is designed to loop through and retrieve vault Object data using VQL endpoint: https://developer.veevavault.com/api/24.2/#submitting-a-query
+def update(configuration: dict, state: dict):
+    base_url = f"https://{configuration.get('vaultDNS')}/api/v24.2/"
```

Comment:
define this as a constant and then use that for all occurrences 
you can keep some regex and then replace that while using in your code with config value.

Reply:
Defined as a constant at the start of the code under #Constant